python_modules/dagster/dagster/_core/definitions/schedule_definition.py

```diff
@@ -113,16 +121,35 @@ def get_or_create_schedule_context(
             raise DagsterInvalidInvocationError(
                 f"Schedule invocation expected argument '{context_param_name}'."
             )
-        context_param_name = context_param_name or list(kwargs.keys())[0]
-        context = check.opt_inst(kwargs.get(context_param_name), ScheduleEvaluationContext)
+        context = (
+            check.opt_inst(kwargs.get(context_param_name), ScheduleEvaluationContext)
+            if context_param_name
+            else check.opt_inst(kwargs.get("context"), ScheduleEvaluationContext)
+        )
     elif context_param_name:
         # If the context parameter is present but no value was provided, we error
         raise DagsterInvalidInvocationError(
             "Schedule evaluation function expected context argument, but no context argument "
             "was provided when invoking."
         )
 
-    return context or build_schedule_context()
+    context = context or build_schedule_context()
+    resource_args_from_kwargs = {}
+
+    resource_args = {param.name for param in get_resource_args(fn)}
+    for resource_arg in resource_args:
+        if resource_arg in kwargs:
+            resource_args_from_kwargs[resource_arg] = kwargs[resource_arg]
+            del kwargs[resource_arg]
+
+    resources_provided_in_multiple_places = (resource_args_from_kwargs) and (context.resource_defs)
+    if resources_provided_in_multiple_places:
```

Comment:
I'm not convinced this should be illegal. In fact in production contexts resources will always be in both the context and available for direct passing, so we are making illegal a perfectly legal state for the program to be in.

Reply:
This is reasonable, changed to support passing both. I am a little confused by

> in production contexts resources will always be in both the context and available for direct passing

I'm trying to conceptualize an instance where a user would would directly invoke a schedule with resources in context & pass directly, in production or otherwise

Reply:
If the code called other code that relies on the resources being attached to the context.